keys/src/hpke.rs

```diff
@@ -0,0 +1,156 @@
+use borsh::{self, BorshDeserialize, BorshSerialize};
+use hpke::{
+    aead::{AeadTag, ChaCha20Poly1305},
+    kdf::HkdfSha384,
+    kem::X25519HkdfSha256,
+    OpModeR,
+};
+use serde::{Deserialize, Serialize};
+
+/// This can be used to customize the generated key. This will be used as a sort of
+/// versioning mechanism for the key.
+const INFO_ENTROPY: &[u8] = b"session-key-v1";
```

Comment:
this is just extra info for generating the derived key fro encryption. So it's more for contextual info. It's useful when we end up using the same key as encryption and signing, but we don't se it can just empty or static like this. I just arbitrarily made it into a versioning scheme